smdebug/core/utils.py

```diff
@@ -102,11 +112,24 @@ def prepare_telemetry_url(processing_job_arn):
 
 
 def setup_profiler_report(processing_job_arn, opt_out=False):
-    # This function is used externally in the profiler report
-    if opt_out is False and bool(processing_job_arn):
-        prepared_telemtry_url = prepare_telemetry_url(processing_job_arn)
+    """
+    This function is used externally in the profiler report
+    We check for the if the env variable "NOTEBOOK_EXECUTION" has been set to determine
+    if this function has been called outside of a notebook environment, such as papermill,
+    in which case, we skip this skip the telemetry request.
+    :param processing_job_arn:
+    :param opt_out: Does not make a telemetry call if opt_out=True
+    Makes a GET request to the prepared telemetry url if opt_out=False
+    :return:
+    """
+    if (
+        opt_out is False
+        and bool(processing_job_arn)
+        and not bool(os.getenv(PAPERMILL_EXECUTION_ENV_VAR))
```

Comment:
Where will this env variable normally be set? I'd suggest we do it here so that it's all managed within this function.

Reply:
This env is set in the by the rule executing papermill.
Why would this fn set it?